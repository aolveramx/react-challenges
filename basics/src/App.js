// import GitHubUsers from "./components/GitHubUsers"

import RegisterForm from "./components/RegisterForm"


const App = () => {
  return (
    <div>
      <RegisterForm />
    </div>
  )
}

export default App
