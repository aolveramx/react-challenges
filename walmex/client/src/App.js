import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom'

import Header from './components/layout/Header'
import Home from './components/Home'
import Users from './components/Users'

import GlobalStyles from './components/styles/Global'
import { Container } from './components/styles/Container.styled'
import { ThemeProvider } from 'styled-components'

const theme = {
  colors: {
    primary: '#336699',
    secondary: '#CC9900',
    light: '#ffffff',
    dark: '#333333',
  },
}

const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <Router>
        <GlobalStyles />
        <Header />
        <Container>
          <Switch>
            <Route exact path='/' component={Home} />
            <Route exact path='/users' component={Users} />
            <Redirect to='/' />
            <Users />
          </Switch>
        </Container>
      </Router>
    </ThemeProvider>
  )
}

export default App
