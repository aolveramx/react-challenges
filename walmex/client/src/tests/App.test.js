import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import App from "../App"

describe('Tests in <App />', () => {

 test('should render <App />', () => {
   const wrapper = shallow(<App />)
   expect(wrapper).toMatchSnapshot()
 });

})