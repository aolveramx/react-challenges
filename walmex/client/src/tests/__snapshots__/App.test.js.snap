// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <App /> should render <App /> 1`] = `
<Component
  theme={
    Object {
      "colors": Object {
        "dark": "#333333",
        "light": "#ffffff",
        "primary": "#336699",
        "secondary": "#CC9900",
      },
    }
  }
>
  <BrowserRouter>
    <Memo(l) />
    <Header />
    <styled.div>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/users"
        />
        <Redirect
          to="/"
        />
        <Users />
      </Switch>
    </styled.div>
  </BrowserRouter>
</Component>
`;
